addons/website_sale_stock/controllers/main.py

```diff
@@ -0,0 +1,25 @@
+# -*- coding: utf-8 -*-
+# Part of Odoo. See LICENSE file for full copyright and licensing details.
+
+from odoo.addons.website_sale.controllers.main import WebsiteSale
+
+
+class WebsiteSale(WebsiteSale):
+
+    def get_attribute_value_ids(self, product):
+        res = super(WebsiteSale, self).get_attribute_value_ids(product)
+        result = []
+        for variant in product.sudo().product_variant_ids:
+            for r in res:
+                if r[0] == variant.id:
```

Comment:
@tde-banana-odoo If the first element is an id then I add a dictionary into it. What do you think ?